rbac/management/group/view.py

```diff
@@ -470,3 +473,34 @@ def roles(self, request, uuid=None):
             return Response(status=status.HTTP_204_NO_CONTENT)
 
         return Response(status=status.HTTP_200_OK, data=response_data.data)
+
+    def obtain_roles(self, request, group):
+        """Obtain roles based on request, supports exclusion."""
+        exclude = self.validate_and_get_exclude_key(request.query_params)
+
+        roles = (group.roles() if exclude == 'false'
+                 else self.obtain_roles_with_exclusion(request, group))
+
+        return [RoleMinimumSerializer(role).data for role in roles]
+
+    def obtain_roles_with_exclusion(self, request, group):
+        """Obtain the queryset for roles based on scope."""
+        scope = request.query_params.get('scope', 'account')
```

Comment:
Good question. As you said, the group is already scoped, either account or principal. So here is just getting the whole set of roles based on this scope of the group. e.g. if a principal is not admin, he could only visit the group that belong to him, then when he call this /groups/<uuid>/roles/, he must specify scope=principal. Because he is not admin, then the whole set of roles should also be scoped as principal. Otherwise, he could see roles beyond his principal.

Reply:
Maybe I'm confused here. 

- it does appear that the `scope=principal` _is_ respected on `GET /groups/<uuid>/` to show a group if I belong to it, is that correct? I don't actually see that documented in our spec, though. so that was unexpected for me. I'm not sure if it _should_ be doing this? we also project the principal information on this endpoint, though I'm not sure that's an issue for users belonging to the group to see.

- assuming the above is true, and fine, then when a user views `/groups/<uuid>/roles/`, the roles are already scoped to the group, which the user has access to, correct? oh! as I was typing, I think what you were conveying clicked. you're saying, when you supply the `exclude=true` filter, _that's_ the case where we can't return all other roles for the principal, we need to scope those (not included in the group), to those which the user has access to?

I'm still not convinced we actually want/need to even allow a principal scope on that exclude filter, since I'm not sure there's a use case for it, but I do think I understand now! 😄 Let me know if I'm off base.

Reply:
Right, you got it, I know it is kind of confusing 👍 Currently, the principal scope is required for granting non-org admin to access their own resource. In the future, are we going to allow non-org admin to access any roles or groups that's not belong to them? If we don't allow that, then the whole set of roles should be scoped to principal. Or they would see roles that's not supposed to be seen by them.

Reply:
That makes sense now, thanks!